src/py/_pyodide/_base.py

```diff
@@ -422,6 +422,30 @@ def eval_code(
         ``None``. If the last statement is an expression, return the result of the
         expression. Use the ``return_mode`` and ``quiet_trailing_semicolon``
         parameters to modify this default behavior.
+
+    Examples
+    --------
+    >>> from pyodide.code import eval_code
+    >>> source = "1 + 1"
+    >>> eval_code(source)
+    2
+    >>> source = "1 + 1;"
+    >>> eval_code(source, quiet_trailing_semicolon=True)
+    >>> eval_code(source, quiet_trailing_semicolon=False)
+    2
+    >>> my_globals = { "y": "100" }
+    >>> my_locals = { "y": "200" }
+    >>> source = "print(locals()['y'], globals()['y'])"
+    >>> eval_code(source, globals=my_globals, locals=my_locals)
+    200 100
+    >>> source = "test = 1 + 1"
+    >>> eval_code(source, return_mode="last_expr_or_assign")
+    2
+    >>> eval_code(source, return_mode="last_expr")
+    >>> eval_code(source, return_mode="none")
+    >>> source = "print('Hello from pyodide')" # Pretend this is open('example_of_filename.py', 'r').read()
+    >>> eval_code(source, filename="example_of_filename.py")
+    Hello from pyodide
```

Comment:
The `filename` argument doesn't actually change the result here in any way. Maybe it would be good to raise an error and look at the traceback?

Reply:
I was thinking the same thing about the result. I've update the example. Let me know if that's what you were thinking as well